### DISCUSSION TITLE
over-fitting

### DISCUSSION THREAD

**Discussion by J-You** [deepmodeling/deepmd-kit#562] [Q&A] [answered]
Hello all,

I trained the data prepared by the script below. However, the train result seems over-fitting.
```
from dpdata import LabeledSystem,MultiSystems
from glob import glob
fs=glob('/home/why/dptrain/raw_data/*.xml')  # remeber to change here !!!
ms=MultiSystems()
for f in fs:
    try:
        ls=LabeledSystem(f, fmt='vasp/xml')
        ls.shuffle()
    except:
        print(f)
    if len(ls)>0:
        ms.append(ls)
ms.to_deepmd_npy(folder='/home/why/dptrain/data', set_size=20000)
```
All of my train parameters are close to the literature, and the trained data size should also be sufficient.
I think my problem is caused by the unreasonable shuffle when preparing data.

Then I create many systems, each of them is distinguished by temperature. The train result of this scenario looks like good.
`systems": ["../data/1500", "../data/1300", "../data/1100", "../data/0900", "../data/0700", "../data/0500"]`

I want to know if there exists a more reasonable solution for different temperature features?
I also want to ask, how to create fparam.npy? I only find the limited description of it in [https://deepmd.readthedocs.io/en/master/train-input.html](url) If it is only suitable to the former version of DP-KIT?

**Comment by njzjz** [comment, marked as answer]
The potential energy is independent from the temperature, meaning you do not need to distinguish them.

**Reply by J-You** [reply]
Hi jinzhe,
Surely as you say, it is meaningless.
I have generated a system with all temperature data by the above dpdata script. But the train  by one system is over-fitting.

![20210424152722](https://user-images.githubusercontent.com/80451798/115951144-bedb4400-a511-11eb-933f-1b49b948080e.png)

I think my problem is caused by the unreasonable shuffle when preparing data.
Do you have any ideas to improve the way of my data preparation?

Thanks for you time and attention.

**Reply by njzjz** [reply]
How much data do you have? I see you set `set_size` to 20000. Do you have multiple sets? If so, the test set will be only the last set and the training set will be others. This means, if you data is in order, such as in time series, the training set and the test set will be different. So shuffle them before divided into training set and test set, if you really want to have a test set.

**Reply by J-You** [reply]
Thanks for your reply.
I have 12,0000 frames for six temperature. I prepare these data as 5 train sets and 1 test set, by setting `set_size` to 20000.
In the script above, it only shuffle one temperature data at once time and then iteratively add each data in `multisystems`.

The correction is to shuffle all of the temperature data at the same time.
I have got it. Thanks a lot.
